version: '3.7'

services:
  veta:
    build:
      context: .
      dockerfile: Dockerfile
    depends_on:
        mysql:
          condition: service_healthy
    networks:
          - veta-network
  mysql:
    image: mysql:8
    environment:
      MYSQL_ROOT_PASSWORD: veta@123
      MYSQL_DATABASE: veta_db
    healthcheck:
      test: [ "CMD-SHELL", "mysqladmin ping -h localhost -u root -p1234" ]
      timeout: 20s
      retries: 10
    ports:
        - "3308:3306"
    networks:
          - veta-network
  milk-rate-service:
    build:
      context: ./milk-rates-manage
      dockerfile: Dockerfile
    depends_on:
      mysql:
        condition: service_healthy
    ports:
      - "9896:9896"
    environment:
      SPRING_DATASOURCE_URL: jdbc:mysql://mysql:3306/veta_db
      SPRING_DATASOURCE_USERNAME: root
      SPRING_DATASOURCE_PASSWORD: veta@123
      SPRING_DATASOURCE_HOST: mysql
    networks:
      - veta-network
  cattle-service:
    build:
      context: ./cattle-manage
      dockerfile: Dockerfile
    depends_on:
      mysql:
        condition: service_healthy
    ports:
      - "9898:9898"
    environment:
      SPRING_DATASOURCE_URL: jdbc:mysql://mysql:3306/veta_db
      SPRING_DATASOURCE_USERNAME: root
      SPRING_DATASOURCE_PASSWORD: veta@123
      SPRING_DATASOURCE_HOST: mysql
    networks:
      - veta-network
networks:
  veta-network:
    driver: bridge
